### PR TITLE
Allow setting output writer for VPN and SSH packages.

### PR DESCRIPTION
**Pull request by Gerrit91**

### CODE REVIEW

File: pkg/ssh/ssh.go [hunk 1]
```diff
@@ -11,8 +12,19 @@ import (
 	"golang.org/x/term"
 )
 
+type ConnectOpt interface{}
```
Comment (majst01) [inline pkg/ssh/ssh.go:15]:
```suggestion
type ConnectOpt any
```

File: pkg/vpn/connect.go [hunk 2]
```diff
@@ -15,6 +16,16 @@ import (
 	"tailscale.com/tsnet"
 )
 
+type ConnectOpt interface{}
```
Comment (majst01) [inline pkg/vpn/connect.go:19]:
```suggestion
type ConnectOpt any
```

File: pkg/vpn/connect.go [hunk 3]
```diff
@@ -15,6 +16,16 @@ import (
...
+
+type connectOptOutputWriter struct {
+	out io.Writer
+}
+
+func ConnectionOptOutputWriter(out io.Writer) ConnectOpt {
```
Comment (majst01) [inline pkg/vpn/connect.go:25]: Maybe 

```suggestion
func WithOutputWriter(out io.Writer) ConnectOpt {
```

File: pkg/ssh/ssh.go [hunk 4]
```diff
@@ -11,8 +12,19 @@ import (
...
+
+type connectOptOutputWriter struct {
+	out io.Writer
+}
+
+func ConnectionOptOutputWriter(out io.Writer) ConnectOpt {
```
Comment (majst01) [inline pkg/ssh/ssh.go:21]: Maybe

```suggestion
func WithOutputWriter(out io.Writer) ConnectOpt {
```

**Review by majst01 — CHANGES_REQUESTED** [review]

File: pkg/ssh/ssh.go (hunk 1 above)
Comment (Gerrit91) [reply pkg/ssh/ssh.go]: Done.

File: pkg/vpn/connect.go (hunk 3 above)
Comment (Gerrit91) [reply pkg/vpn/connect.go]: The `With...` functions I typically use for chaining constructors, but this is a bit different as options are not meant to be chained, so I find it a bit misleading.

File: pkg/vpn/connect.go (hunk 2 above)
Comment (Gerrit91) [reply pkg/vpn/connect.go]: Done.

**Review by majst01 — APPROVED** [review]

**Pull request merged**